web/src/beta/features/PluginPlayground/Code/hook.ts

```diff
@@ -0,0 +1,110 @@
+import Visualizer from "@reearth/beta/features/Visualizer";
+import * as yaml from "js-yaml";
+import { ComponentProps, useCallback, useState } from "react";
+
+import { WidgetLocation } from "../../Visualizer/Crust/Widgets/types";
+import { FileType } from "../Plugins/constants";
+
+type Widgets = ComponentProps<typeof Visualizer>["widgets"];
+
+type ReearthYML = {
+  id: string;
+  name: string;
+  version: string;
+  extensions?: {
+    id: string;
+    type: string;
+    name: string;
+    description: string;
+    widgetLayout?: {
+      defaultLocation: {
+        zone: WidgetLocation["zone"];
+        section: WidgetLocation["section"];
+        area: WidgetLocation["area"];
+      };
+    };
+  }[];
+};
+
+type Props = {
+  files: FileType[];
+};
+
+const getYmlJson = (file: FileType) => {
+  try {
+    return yaml.load(file.sourceCode) as ReearthYML;
+  } catch (error) {
+    if (error instanceof yaml.YAMLException) {
+      console.error(error.message);
```

Comment:
Maybe we can add an error notification for the user using `useNotification` if an error occurs?

Reply:
fixed this: 

https://github.com/reearth/reearth-visualizer/pull/1224/commits/1bec19edc93e2cebc4f89b876c578439a63a394f